### PR TITLE
E2E tests for SQS splitting

### PR DESCRIPTION
**Pull request by t4lz**
Part of #2670 

This PR adds 1 E2E test for a basic SQS splitting case. There are still things missing, and some things that are potential issue either in the mirrord-SQS-operator or in the test, but I would like to merge the test and solve more issues later, so that we already have basic protection against regressions.

For the test there should be a deployed localstack instance. A test fixture patches the installed operator to use that localstack instead of actual AWS SQS.
The test deploys a service that consumes 2 queues (one normal, one FIFO), starts two clients that split messages with different filters, sends messages with attributes that make some of them reach each of the two clients and the deployed application. The test then verifies each of those 3 applications received the messages it was supposed to.

Unresolved issues:
- Relying on sleeps instead of providing/identifying a reliable indicator that filtering started.
- There is no verification that all queues are automatically removed after all sessions are done, because that did not work - might be a bug.
- Support for running the test with actual SQS instead of localstack.

https://github.com/metalbear-co/test-images/pull/21 has to be merged first, then the image can be changed to the published test image.

### CODE REVIEW

**Review by DmitryDodzin — COMMENTED** [review]
Maybe move second docker to second repo aswell

File: tests/src/utils.rs [hunk 1]
```diff
@@ -1429,28 +1555,42 @@ pub fn resolve_node_host() -> String {
...
-        .unwrap();
-    if host_ip.parse::<Ipv4Addr>().unwrap().is_private() {
-        host_ip = resolve_node_host();
-    }
-    let services_api: Api<Service> = Api::namespaced(kube_client.clone(), &service.namespace);
+        .and_then(|ip| {
+            ip.parse::<Ipv4Addr>()
+                .unwrap()
+                .is_private()
+                .not()
```
[the remark this answers is not in the document]
Comment (t4lz) [reply tests/src/utils.rs:1579]: Could you please elaborate? :)

File: tests/rust-sqs-printer/sqs-printer.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      app: sqs-printer
+  template:
+    metadata:
+      labels:
+        app: sqs-printer
+    spec:
+      serviceAccountName: sqs-reader-account
+      containers:
+        - name: sqs-printer
+          image: docker.io/t4lz/sqs-printer:latest
```
[the remark this answers is not in the document]
Comment (t4lz) [reply tests/rust-sqs-printer/sqs-printer.yaml:20]: On CI or just once? This K8s manifest and Dockerfile are for manual testing, and are not used in the E2E tests, btw.

File: tests/src/utils.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DmitryDodzin) [reply tests/src/utils.rs:1579]: I'm `self.is_fan().not()` but not entirely against this

File: tests/rust-sqs-printer/sqs-printer.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (DmitryDodzin) [reply tests/rust-sqs-printer/sqs-printer.yaml:20]: Ok =]

File: tests/rust-sqs-printer/Cargo.toml
[the remark this answers is not in the document]
Comment (DmitryDodzin) [reply tests/rust-sqs-printer/Cargo.toml:1]: Ok I just saw the Dockerfile + platform.sh and assumed it's another app

**Review by DmitryDodzin — APPROVED** [review]
LGTM

**Pull request merged**